### PR TITLE
Add security levels

### PR DESCRIPTION
**Pull request by iam-py-test**
Is this look ok?

### CODE REVIEW

File: ref/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from types import ModuleType
 import importlib
-def main(__builtins__: ModuleType) -> None:
+def main(__builtins__: ModuleType, restrictwrite: bool = False) -> None:
```
Comment (iam-py-test) [inline ref/__init__.py]: Is this correct?

File: ref/__init__.py [hunk 2]
```diff
@@ -23,4 +23,9 @@ def importer(name, *args, level: int = 0):
             except AttributeError: pass
     if level >= 1:
         pass # todo: add here some other functions
+    if restrictwrite == True:
```
Comment (iam-py-test) [inline ref/__init__.py]: This was just copy-pasted from https://github.com/donno2048/restricted-functions/issues/1#issuecomment-863331876

**Comment by iam-py-test** [comment]
Is this PR look ok?

**Comment by donno2048** [comment]
Did you test it to see if it works? And anyway didn't we agree to use the flag method?

**Comment by iam-py-test** [comment]
Sorry for my error. Let me undo my changes
Was the original ok?

**Comment by iam-py-test** [comment]
Corrected. I totally misunderstood your first comment

**Comment by donno2048** [comment]
Ok, first a shorter name could be better (not required), secondly, you need to test it (clone the repo and checkout to this branch then `pip install .`) and lastly, I'm going on a vacation and will be back on Sunday, you can keep working on it but don't have to, see you soon...

**Comment by iam-py-test** [comment]
> Ok, first a shorter name could be better (not required), secondly, you need to test it (clone the repo and checkout to this branch then `pip install .`) and lastly, I'm going on a vacation and will be back on Sunday, you can keep working on it but don't have to, see you soon...

Ok. Hope you have a nice vacation. 
I will continue adding functions but won't make any major changes

File: test.py [hunk 3]
```diff
@@ -19,3 +19,6 @@
 try: shutil.rmtree("../restricted-functions")
 except: pass
 else: raise Exception("failed")
+try: open("w", "setup.py").write("text")
+except: pass
+else: raise Exception("failed")
```
Comment (iam-py-test) [inline test.py:21]: @donno2048 maybe this should be replaced with a call to `print` to the rest of the program can be tested
```suggestion
else: print("Failed to block write")
```
Edit: Sorry for my typo

File: test.py [hunk 4]
```diff
@@ -19,3 +19,6 @@
 try: shutil.rmtree("../restricted-functions")
 except: pass
 else: raise Exception("failed")
+try: open("w", "setup.py").write("text")
```
Comment (iam-py-test) [inline test.py]: Maybe replace this with this to prevent an overwrite of a critical file:
```suggestion
try: open("w", "writeme.txt").write("text")
```

**Review by iam-py-test — COMMENTED** [review]
Tested on Linux and it worked to block write

**Comment by iam-py-test** [comment]
@donno2048 is this ready to merge?

File: TODO.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 - [ ] Get more dangerous functions
-  - [ ] os.unlink
+  - [X] os.unlink
```
Comment (iam-py-test) [inline TODO.md:2]: Updated as os.unlink has been added

Comment (iam-py-test) [inline TODO.md:2]: Updated as os.unlink has been added

**Comment by donno2048** [comment]
I'll make the check for `restrictwrite` inside the open function, I think

File: TODO.md [hunk 6]
```diff
@@ -20,6 +20,7 @@
   - [ ] os.kill
   - [ ] os.killpg
   - [ ] os.plock
+  - [ ] os.removedirs
   - [ ] execfile
```
Comment (donno2048) [inline TODO.md]: Why?

Comment (donno2048) [reply TODO.md]: `execfile` is just more python....

Comment (iam-py-test) [reply TODO.md]: It is for executing other files; that could be dangerous.

Comment (donno2048) [reply TODO.md]: I know, but wouldn't all the imports will be executed in the main module level of the original process?

Comment (iam-py-test) [reply TODO.md]: > I know, but wouldn't all the imports will be executed in the main module level of the original process?

I guess...
I removed it anyway. (someone could probably use something else to do the same thing - `eval(open("file.py").read())`?)

Comment (donno2048) [reply TODO.md]: Sure... Using `exec` makes more sense, but you're right (if it's actually in the main module level of the original process)

Comment (donno2048) [reply TODO.md]: Btw, `execfile` was removed in python3 and now you suppose to use `exec` with `open` anyways

**Pull request merged**